Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/ChillerStage.mo

```diff
@@ -0,0 +1,234 @@
+  connect(chiTwo.y,y[2])
+    annotation (Line(points={{142,-10},{150,-10},{150,5},{170,5}},color={255,0,255}));
+  connect(on,offToOne.condition)
+    annotation (Line(points={{-180,60},{-50,60},{-50,40},{-2,40}},color={255,0,255}));
+  connect(oneToTwo.condition,thrOneToTwo.y)
+    annotation (Line(points={{-2,-40},{-14,-40}},color={255,0,255}));
+  connect(thrTwoToOne.y,twoToOne.condition)
+    annotation (Line(points={{-15,-74},{-6,-74},{-6,-56},{40,-56},{40,-40},{48,-40}},color={255,0,255}));
+  connect(on,notOn.u)
+    annotation (Line(points={{-180,60},{-50,60},{-50,20},{-32,20}},color={255,0,255}));
+  connect(notOn.y,oneToOff.condition)
+    annotation (Line(points={{-8,20},{28,20},{28,40}},color={255,0,255}));
+  connect(dT.y,pro.u1)
+    annotation (Line(points={{-129,6},{-112,6}},color={0,0,127}));
+  connect(cp.u,pro.y)
+    annotation (Line(points={{-82,0},{-89,0}},color={0,0,127}));
+  connect(cp.y,thrOneToTwo.u)
+    annotation (Line(points={{-59,0},{-50,0},{-50,-40},{-38,-40}},color={0,0,127}));
+  connect(dT.u1,TChiWatRet)
+    annotation (Line(points={{-152,12},{-156,12},{-156,20},{-180,20}},color={0,0,127}));
+  connect(TChiWatSup,dT.u2)
+    annotation (Line(points={{-180,-20},{-156,-20},{-156,0},{-152,0}},color={0,0,127}));
+  connect(pro.u2,mFloChiWat)
+    annotation (Line(points={{-112,-6},{-120,-6},{-120,-60},{-180,-60}},color={0,0,127}));
+  connect(oneToTwo.condition, thrTwoToOne.u) annotation (Line(points={{-2,-40},
+          {-10,-40},{-10,-54},{-50,-54},{-50,-74},{-38,-74}}, color={255,0,255}));
+  annotation (
+    defaultComponentName="chiStaCon",
+    Diagram(
+      coordinateSystem(
+        preserveAspectRatio=false,
+        extent={{-160,-100},{160,100}})),
+    Icon(
+      coordinateSystem(
+        preserveAspectRatio=false,
+        extent={{-100,-100},{100,100}}),
+      graphics={
+        Text(
+          extent={{-150,150},{150,110}},
+          textString="%name",
+          lineColor={0,0,255})}),
+    Documentation(
+      revisions="<html>
+<ul>
+<li>
+August 6, 2020 by Jing Wang:<br/>
+First implementation.
+</li>
+</ul>
+</html>",
+      info="<html>
+<p>This model implements the staging control logic as follows: </p>
+<ul>
+<li>When the plant enabling signal <code>on</code> changes from <code>false</code> 
+to <code>true</code>, one chiller is enabled. </li>
+<li>When the total cooling load <code>QLoa</code> exceeds 80 percent (adjustable) 
+of one chiller&apos;s nominal capacity <code>QChi_nominal</code>, a second chiller 
+is enabled. </li>
+<li>When the total cooling load <code>QLoa</code> drops below 60 percent 
+(adjustable) of one chiller&apos;s nominal capacity <code>QChi_nominal</code> 
+(i.e., 30 percent each chiller), the second chiller is disabled. </li>
+<li>When the plant enabling signal <code>on</code> changes from <code>true</code> 
+to <code>false</code>, the operating chiller is disabled.</li>
```

Comment:
Thank you for your question. Actually, we designed the on signal to represent the plant operator. In the real-world operation of a DC plant, it is very unlikely that the cooling load would drop from a high level (e.g., 2 chillers) to none - unless it's under emergency. So we did not design the model for that case. Let us know if you think a model refactoring is needed to include that case. Thanks!

Reply:
> Thanks for the explanation. I think I am still a bit confused:
> 
> * If the `on` signal comes from the plant operator, then it is not necessarily tied to low/high loads. I would agree that a plant operator is unlikely to shut off the plant when there is a high load demand, but I don't think this assumption should be hardcoded in how the chillers are staged.
> * Conversely, I think this model is also missing a cut-off load, under which the plant goes from "One Chiller" to "Off". I originally assumed that this signal was the `on` flag, but given your definition, I think they are separate. Put differently, it is possible for the plant to be `on`, but for the load to be too small to require any chiller to run.
> 
> My suggestion would be to separate the stating sequence from the `on` signal. Something like:
> 
> * If `on` is false, then all chillers are off (regardless of load or chiller current operation)
> * If `on` is true, then run the staging to determine what chillers are on.

Thanks for your suggestion. We added one more transitional condition to reflect the situation of “the signal turns to false when two chillers are on”. The clauses in [ASHRAE Guideline 36-2018](https://www.ashrae.org/file%20library/technical%20resources/standards%20and%20guidelines/standards%20addenda/g36_2018_x_20210709.pdf), from 5.20.4.19 to 5.20.4.35, mentioned that “whenever there is a stage-down command, shut off the last stage chiller”. So, we added this to the state graph and the model implementation to sequentially shutting off the chillers. 
![image](https://user-images.githubusercontent.com/82961137/129104839-e29ee33b-914c-4876-ae80-326c21d66bb5.png)
In terms of the cut-off load, I discussed with Dr.Zuo and he suggested we keep the current control logic because even if the load is small, it still needs to be met. Also, the chiller performance curve actually has a parameter of minimum load, which I assume should work for the case of a very small load.